src/package/cppwinrt/nuget/Microsoft.Windows.CppWinRT.targets

```diff
@@ -243,19 +257,37 @@ Copyright (C) Microsoft Corporation. All rights reserved.
         <Message Text="CppWinRTMdMergeMetadataDirectories: @(CppWinRTMdMergeMetadataDirectories)" Importance="$(CppWinRTVerbosity)"/>
     </Target>
 
-    <!--If building any Xaml content, include XamlMetaDataProvider in project metadata-->
-    <Target Name="CppWinRTAddXamlMetaDataProviderIdl" Condition="'$(XamlLanguage)' == 'CppWinRT' and '$(CppWinRTAddXamlMetaDataProviderIdl)' == 'true'">
+    <!-- Adds the XamlMetadataProvider idl to the Midl itemgroup, if building any xaml content -->
+    <Target Name="CppWinRTComputeXamlGeneratedMidlInputs"
+            DependsOnTargets="$(CppWinRTComputeXamlGeneratedMidlInputsDependsOn)"
+            Condition="'@(Page)@(ApplicationDefinition)' != '' and '$(XamlLanguage)' == 'CppWinRT'">
         <PropertyGroup>
-            <XamlMetaDataProviderIdl>
-                $(IntDir)XamlMetaDataProvider.idl
-            </XamlMetaDataProviderIdl>
-            <XamlMetaDataProviderCpp>
-                $(GeneratedFilesDir)XamlMetaDataProvider.cpp
-            </XamlMetaDataProviderCpp>
-            <_HasXaml>false</_HasXaml>
-            <_HasXaml Condition="'@(Page)' != '' Or '@(ApplicationDefinition)' != ''">true</_HasXaml>
+            <_DisableReferences>false</_DisableReferences>
```

Comment:
@Scottj1s I am not 100% sure if I understood the _DisableReferences property. Can you double check that I refactored that the right way?

Reply:
I think so, but there's some clean up needed below (_DisableReferences references)

---
In reply to: [328949800](https://github.com/microsoft/xlang/pull/609#discussion_r328949800) [](ancestors = 328949800)